### PR TITLE
Require socket in the file that needs it

### LINKED ISSUE #5533: uninitialized constant Bundler::MirrorSockets::Socket
# Error Report

## Questions

Please fill out answers to these questions, it'll help us figure out
why things are going wrong.

- **What did you do?**

  I ran the command `bundle outdated`

- **What did you expect to happen?**

  Show me the output of the `bundle outdated` command.

- **What happened instead?**

   uninitialized constant Bundler::MirrorSockets::Socket

- **Have you read our issues document, https://github.com/bundler/bundler/blob/master/doc/contributing/ISSUES.md?**

Yes

## Backtrace

```
NameError: uninitialized constant Bundler::MirrorSockets::Socket
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/mirror.rb:185:in `initialize'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/mirror.rb:145:in `new'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/mirror.rb:145:in `replies?'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/mirror.rb:102:in `validate!'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/mirror.rb:46:in `fetch_valid_mirror_for'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/mirror.rb:23:in `for'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/settings.rb:133:in `mirror_for'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/source/rubygems/remote.rb:10:in `initialize'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/source/rubygems.rb:234:in `new'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/source/rubygems.rb:234:in `block in fetchers'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/source/rubygems.rb:233:in `map'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/source/rubygems.rb:233:in `fetchers'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/source/rubygems.rb:346:in `block in remote_specs'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/index.rb:10:in `build'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/source/rubygems.rb:345:in `remote_specs'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/source/rubygems.rb:82:in `specs'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/definition.rb:263:in `block (2 levels) in index'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/definition.rb:261:in `each'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/definition.rb:261:in `block in index'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/index.rb:10:in `build'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/definition.rb:258:in `index'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/definition.rb:252:in `resolve'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/definition.rb:176:in `specs'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/definition.rb:164:in `resolve_remotely!'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/cli/outdated.rb:52:in `block in run'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/cli/outdated.rb:58:in `run'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/cli.rb:288:in `outdated'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/cli.rb:20:in `dispatch'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `start'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/cli.rb:11:in `start'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/exe/bundle:32:in `block in <top (required)>'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/lib/bundler/friendly_errors.rb:121:in `with_friendly_errors'
  /Users/rafaelfranca/.gem/ruby/2.3.3/gems/bundler-1.14.6/exe/bundle:24:in `<top (required)>'
  /Users/rafaelfranca/.gem/ruby/2.3.3/bin/bundle:22:in `load'
  /Users/rafaelfranca/.gem/ruby/2.3.3/bin/bundle:22:in `<main>'
```

## Environment

```
Bundler   1.14.6
Rubygems  2.6.10
Ruby      2.3.3p222 (2016-11-21 revision 56859) [x86_64-darwin16]
GEM_HOME  /Users/rafaelfranca/.gem/ruby/2.3.3
GEM_PATH  /Users/rafaelfranca/.gem/ruby/2.3.3:/opt/rubies/2.3.3/lib/ruby/gems/2.3.0
Git       2.12.1
Platform  x86_64-darwin-16
```

## Bundler settings

```
jobs
  Set for your local app (/Users/rafaelfranca/src/shopify/shopify/.bundle/config): "4"
  Set for the current user (/Users/rafaelfranca/.bundle/config): "4"
build.gda
  Set for your local app (/Users/rafaelfranca/src/shopify/shopify/.bundle/config): "--with-cflags=-I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/libxml2"
github.https
  Set for your local app (/Users/rafaelfranca/src/shopify/shopify/.bundle/config): "true"
mirror.https://rubygems.org/
  Set for your local app (/Users/rafaelfranca/src/shopify/shopify/.bundle/config): "https://my-custom.proxy"
mirror.http://rubygems.org/
  Set for your local app (/Users/rafaelfranca/src/shopify/shopify/.bundle/config): "https://my-custom.proxy"
mirror.https://rubygems.org.fallback_timeout/
  Set for your local app (/Users/rafaelfranca/src/shopify/shopify/.bundle/config): "1"
mirror.http://rubygems.org.fallback_timeout/
  Set for your local app (/Users/rafaelfranca/src/shopify/shopify/.bundle/config): "1"
```

### PR DESCRIPTION
**Pull request by rafaelfranca**
Closes #5533

### CODE REVIEW

**Comment by rafaelfranca** [comment]
I'd love ideas on how to test it is necessary.

**Comment by andremedeiros** [comment]
Something similar happened to me on #4083 and I didn't add a test.

Not sure if it's necessary (given that the file actually needs `Socket` anyways.)

File: lib/bundler/mirror.rb [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # frozen_string_literal: true
+
```
Comment (andremedeiros) [inline lib/bundler/mirror.rb]: Unnecessary newline

File: lib/bundler/mirror.rb [hunk 2]
```diff
@@ -1,4 +1,7 @@
...
+require 'socket'
```
Comment (andremedeiros) [inline lib/bundler/mirror.rb]: Can you use double quotes here? (inline with the rest of this codebase)

**Comment by andremedeiros** [comment]
@bundlerbot r+

**Pull request merged**